### ISSUE TITLE
Catalogs - helm stable catalog is disbaled but charts are still showing on the apps catalog

### ISSUE THREAD

**Issue by guangbochen** [rancher/rancher#12865]
Rancher versions: 2.0 master 04/019
Issue: Helm stable catalog is disbaled but the charts are still showing on the apps catalog page

Steps to Reproduce:
1. disable the `Helm Stable` catalog from the Catalog page
2. after a couple of minutes the stable charts are still showing on the `catalog apps` page, but the v3/catalog?catalog=helm is already removed.
![image](https://user-images.githubusercontent.com/4569037/38985033-da4164e8-43fa-11e8-8044-ae40b3e40cbc.png)

![image](https://user-images.githubusercontent.com/4569037/38986029-bc0e0366-43fd-11e8-9e82-cbe8c910cf8d.png)

etcd error:
```
2018-04-19 10:12:56.167148 W | etcdserver: avoid queries with large range/delete range!
2018-04-19 10:12:57.092914 I | mvcc: finished scheduled compaction at 33958 (took 924.400929ms)
2018-04-19 10:12:58.490287 W | etcdserver: apply entries took too long [471.637795ms for 1 entries]
2018-04-19 10:12:58.490445 W | etcdserver: avoid queries with large range/delete range!
```

Expected Results: charts should be removed if catalogs are disabled.

**Comment by StrongMonkey** [comment]
for templates they took a while to remove. Does it still exist after 5 minutes?

**Comment by guangbochen** [comment]
yep, it still exists and I think it was caused by the etcd server with `etcdserver: avoid queries with large range/delete range!`

**Comment by guangbochen** [comment]
verified fix on rancher/rancher:master 04.27, and close it.
